src/Lesson5/Computer.java

```diff
@@ -0,0 +1,96 @@
+    static Random random = new Random(); //статичесткие чтобы видеть по всему класу
+    static Scanner scanner = new Scanner(System.in);
+    boolean procesor; //поля екзепляров
+    boolean operativka;
+    boolean zorstkijDisk;
+    int resursPolnogoCykluRaboty;
+    boolean isOn = false; //даю по умолчанию false чтобы пометять когда включается
+    boolean isDead = false; //даю по умолчанию false чтобы пометять когда сгорает компьютер
+
+
+    //перегружений конструктор
+    public Computer(boolean procesor, boolean operativka, boolean zorstkijDisk, int resursPolnogoCykluRaboty) {
+        this.procesor = procesor;
+        this.operativka = operativka;
+        this.zorstkijDisk = zorstkijDisk;
+        this.resursPolnogoCykluRaboty = resursPolnogoCykluRaboty;
+    }
+
+    //метод который выводить инфо про компьютер на консоль
+    void opisanie() {
+        System.out.print("[");
+        System.out.print("Процесор: " + procesor + ", оперативка: " + operativka + ", жесткий диск: " + zorstkijDisk + ", ресурс полных циклов работы: " + resursPolnogoCykluRaboty);
+        System.out.print("]\n");
+    }
+
+    //проверяем на сбой
+    boolean proverkaNaZboj() {
+        int randomNum = random.nextInt(2); //получаем 0 или 1 случайно
+        System.out.print("Введите число 0 или 1 -> ");
+        int userNum = scanner.nextInt();               //с клавиатури получаем 0 или 1
+        if (randomNum == userNum) {                   //если рандмное число и число пользователя совпадает вызов метода включить компьютер
+            System.out.println("Проверка прошла успешно!");
+            isDead = false;
+        } else {                                                      //если рандмное число и число пользователя не совпадает
+            resursPolnogoCykluRaboty = 0;                    //обнуляем цикли
+            isDead = true;                                               //компьютер умер потому что не прошел проверку
+            System.out.print("Компьютер сгорел. ");
+            System.out.println(resursPolnogoCykluRaboty + "-> остаточный ресурс полного циклу компьютера.");
+        }
+        return isDead;
+    }
+
+    boolean vkluchytComputer() {
+        proverkaNaZboj();  //вызов метода проверки
+        if (procesor & operativka & zorstkijDisk & resursPolnogoCykluRaboty > 0 & (!isOn) & (!isDead)) {//проверяем все ли исправно
+            isOn = true;                   //включаем компьютер
+            System.out.println("Вы включили компьютер!");
+        } else if (resursPolnogoCykluRaboty == 0 & (!isDead)) {   //проверка успешна, но циклы закончились поэтому умер
+            isDead = true; //компьютер умер
+            isOn = false;  // не может быть включен
+            System.out.println("Превышение лимита ресурса, компьютер умер :(");
+        } else if (isOn) {
+            System.out.println("Компьютер уже был включен!");
+        } else {
+            System.out.println("Невозможно включить компьютер :(");
+        }
```

Comment:
@Nadya101 Совет, если тебе много условий проверить надо, чтобы все они выполнялись, то лучше проверять обратное им условие, чтобы хоть один не совпадал
```suggestion
       if (!procesor || !operativka || !zorstkijDisk || resursPolnogoCykluRaboty > 0 || (!isOn) & (!isDead)) {//проверяем если хоть что-то не исправно
            .......
            Далее вся логика переделывается уже исходя из того, что мы сначала проверили на поломки, а не 
            на исправность всего сразу
```
Ты также использовала `&`, а нодо было `&&` (1 - складывает 2 значения, а второй добавляет еще 1 условие с помет `И`.

Reply:
@nickrogracher вот здесь не поняла. Переделала так: 
  _boolean onComputer() {
        CheckingForFailure();
        if (!cpu || !ram || !hdd || maxNumberOfShutdowns == 0 || (isDead)) {//проверяем если хоть что-то не исправно
            System.out.println("Невозможно включить компьютер :(");
            isOn = false;
            getDescription();
        } else if (isOn) {
            System.out.println("Компьютер уже был включен!");
        } else {
            isOn = true;
            System.out.println("Компьютер включен!");
        }
        return isOn;_